metricbeat/autodiscover/builder/hints/metrics.go

```diff
@@ -175,7 +179,12 @@ func (m *metricHints) getHostsWithPort(hints common.MapStr, port int) []string {
 		}
 	}
 
-	return result
+	if len(thosts) > 0 && len(result) == 0 {
+		logp.Debug("hints.builder", "no hosts selected for port %d with hints: %+v", port, thosts)
```

Comment:
Should we make the switch to m.logger here instead of logp? 😬not sure if putting them in same pr is good or not.

Reply:
This module doesn't have a logger yet, I'd leave this for a future change.